src/components/Add.js

```diff
@@ -4,29 +4,56 @@ function Add() {
 
     const [foodName, setName] = useState("");
     const [foodCalories, setCalories] = useState('');
-    const [foodImage, setImage] = useState('');
+    const [foodImage, setImage] = useState('https://cdn.iconscout.com/icon/free/png-256/fast-food-1851561-1569286.png');
 
     const [showForm, setShowForm] = useState(false);
 
+    //checks when state of form is set to false/true
     const showMyForm = (e) => {
         e.preventDefault()
         setShowForm(!showForm);
     };
 
+    // submit button handling - code below can be used, needs refactoring
+    // add props + onSubmit={handleSubmit}
+    // const handleSubmit = (e) => {
+    //     e.preventDefault();
+    //     props.setFood((prevFood) => {
+    //       const newFood = {
+    //         name: name,
+    //         calories: calories,
+    //         image: image,
+    //       };
+    //       return [newFood, ...prevFood];
+    //     });
+
+
+        // target values for each property
+        const handleSetName = (e) => {
+        setName(e.target.value);
+        };
+        const handleSetCalories = (e) => {
+        setCalories(e.target.value);
+        };
+        const handleSetImage = (e) => {
+        setImage(e.target.value);
+        };
+
     return (
         <div>
         <form>
             <button onClick={showMyForm}>Add new food</button>
         </form>
         {showForm && (
             <form>
```

Comment:
One very important step with all the forms in react -> you have to handle the submission, do something with the result right? So for example:

```
 <form onSubmit={handleSubmit}>
```

```
  const handleSubmit = (e) => {
    e.preventDefault();

// at this point you have updated the state already
    const newFood = { foodName, foodCalories,  foodImage }; 

// here we call the function of the paretn component and sent new object to be added to original array 
// you need to create that function addFood in App.js
    props.addFood(newFood);

// here we reset the form input fields to empty values
    setName('');
    setCalories('');
    setImage('');
  };
```